src/daft-parquet/src/stream_reader.rs

```diff
@@ -239,6 +350,153 @@ pub(crate) async fn local_parquet_read_async(
     recv.await.context(super::OneShotRecvSnafu {})?
 }
 
+#[allow(clippy::too_many_arguments)]
+pub(crate) fn local_parquet_stream(
+    uri: &str,
+    original_columns: Option<Vec<String>>,
+    columns: Option<Vec<String>>,
+    start_offset: Option<usize>,
+    original_num_rows: Option<usize>,
+    num_rows: Option<usize>,
+    row_groups: Option<Vec<i64>>,
+    predicate: Option<ExprRef>,
+    schema_infer_options: ParquetSchemaInferenceOptions,
+    metadata: Option<Arc<parquet2::metadata::FileMetaData>>,
+    maintain_order: bool,
+) -> DaftResult<(
+    Arc<parquet2::metadata::FileMetaData>,
+    BoxStream<'static, DaftResult<Table>>,
+)> {
+    let chunk_size = 128 * 1024;
+    let (metadata, schema_ref, row_ranges, column_iters) = local_parquet_read_into_column_iters(
+        uri,
+        columns.as_deref(),
+        start_offset,
+        num_rows,
+        row_groups.as_deref(),
+        predicate.clone(),
+        schema_infer_options,
+        metadata,
+        chunk_size,
+    )?;
+
+    // Create a channel for each row group to send the processed tables to the stream
+    // Each channel is expected to have a number of chunks equal to the number of chunks in the row group
+    let (senders, receivers): (Vec<_>, Vec<_>) = row_ranges
+        .iter()
+        .map(|rg_range| {
+            let expected_num_chunks =
+                f32::ceil(rg_range.num_rows as f32 / chunk_size as f32) as usize;
+            tokio::sync::mpsc::channel(expected_num_chunks)
+        })
+        .unzip();
+    // Create a channel to send errors to the stream
```

Comment:
you shouldn't need to have a seperate channel to send errors. Normally we can just send `DaftResult` into the output channel.